email/fetcher.go

```diff
@@ -313,27 +313,19 @@ func (f *Fetcher) persistMessage(mailbox *imap.MailboxStatus, msg *imap.Message,
 		return errors.AddContext(err, "could not read msg body")
 	}
 
-	// parse the 'from', we evaluate the reply-to first because 'abuse@' emails
-	// are forwarded to our mailbox and the original sender is in the reply-to
-	// field, while the from field holds 'abuse@', so we lose the original
-	// context
-	from := "unknown"
-	if len(msg.Envelope.ReplyTo) > 0 {
-		from = msg.Envelope.ReplyTo[0].Address()
-	} else if len(msg.Envelope.From) > 0 {
-		from = msg.Envelope.From[0].Address()
-	}
-
 	// create the email entity from the message
 	email := database.AbuseEmail{
 		ID:        primitive.NewObjectID(),
 		UID:       uid,
 		UIDRaw:    msg.Uid,
 		Body:      body,
-		From:      from,
 		Subject:   msg.Envelope.Subject,
 		MessageID: msg.Envelope.MessageId,
 
+		From:    extractField("From", msg.Envelope),
+		ReplyTo: extractField("ReplyTo", msg.Envelope),
+		To:      extractField("To", msg.Envelope),
```

Comment:
We don't check any of the parameters(`msg`, `mailbox`, `section`) for being `nil`. This could cause a panic.